### ISSUE TITLE
Not able to target remote minions

### ISSUE THREAD

**Issue by ooCHUBSoo** [StackStorm-Exchange/stackstorm-salt#11]
Hello, 

I am trying to use the stackstorm with Salt integration. I am not able to target systems using api or gui. When I try using test.ping it works as expected however, a grains.get does not accept my target host or my `args=os`:

```sh
[root@#### ~]# st2 --version
st2 2.10.4, on Python 2.7.5
```

`test.ping` result
```sh
[root@#### ~]# st2 run salt.local module=test.ping target=####
.
id: 5cb79dcb94482e74ed7dd0a8
status: succeeded
parameters: 
  module: test.ping
  target: ####
result: 
  exit_code: 0
  result:
    return:
    - ####: true
  stderr: 'st2.actions.python.SaltLocal: INFO     [salt] Payload to be sent: {''username'': ''************'', ''expr_form'': u''glob'', ''eauth'': u''pam'', ''tgt'': u''####'', ''fun'': u''test.ping'', ''password'': ''********'', ''client'': ''local''}
    st2.actions.python.SaltLocal: INFO     [salt] Request generated
    st2.actions.python.SaltLocal: INFO     [salt] Preparing to send
    '
  stdout: ''
```

Results of `grains.get`:

```sh
[root@#### ~]# st2 run salt.local module=grains.get args=os target=#### 
id: 5cb79e7c94482e74ed7dd0ab
status: failed
parameters: 
  args:
  - os
  module: grains.get
  target: ####
result: 
  exit_code: 1
  result: None
  stderr: "st2.actions.python.SaltLocal: INFO     [salt] Payload to be sent: {'username': '************', 'tgt': u'####', 'client': 'local', 'expr_form': u'glob', 'eauth': u'pam', 'arg': [u'os'], 'fun': u'grains.get', 'password': '********'}
st2.actions.python.SaltLocal: INFO     [salt] Request generated
st2.actions.python.SaltLocal: INFO     [salt] Preparing to send
Traceback (most recent call last):
  File "/opt/stackstorm/st2/lib/python2.7/site-packages/python_runner/python_action_wrapper.py", line 325, in <module>
    obj.run()
  File "/opt/stackstorm/st2/lib/python2.7/site-packages/python_runner/python_action_wrapper.py", line 184, in run
    output = action.run(**self._parameters)
  File "/opt/stackstorm/packs/salt/actions/local.py", line 54, in run
    return resp.json()
  File "/opt/stackstorm/st2/lib/python2.7/site-packages/requests/models.py", line 885, in json
    return complexjson.loads(self.text, **kwargs)
  File "/opt/stackstorm/st2/lib/python2.7/site-packages/simplejson/__init__.py", line 518, in loads
    return _default_decoder.decode(s)
  File "/opt/stackstorm/st2/lib/python2.7/site-packages/simplejson/decoder.py", line 370, in decode
    obj, end = self.raw_decode(s)
  File "/opt/stackstorm/st2/lib/python2.7/site-packages/simplejson/decoder.py", line 400, in raw_decode
    return self.scan_once(s, idx=_w(s, idx).end())
simplejson.errors.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
"
  stdout: ''
```

Looks like it's missing an argument but not sure where to add this. Seems more likely a syntax issue on my end.

**Comment by LindsayHill** [comment]
Looks like that module is not returning valid JSON. You could try adding some additional logging around [here](https://github.com/StackStorm-Exchange/stackstorm-salt/blob/master/actions/local.py#L53) to see what it's getting back. 

You could also use the `salt.local_grains.get` action, although it may hit the same underlying issue.

**Comment by ooCHUBSoo** [comment]
> You could also use the `salt.local_grains.get` action, although it may hit the same underlying issue.

Same result. I think it may have something to do with a custom function I am using in salt. I will troubleshoot and update...

Quick questions - 
1. What is the correct syntax if I wanted to use salt.local pack to target specific hosts or hosts by grains?
I tried the following but not sure it is the correct syntax:
`[root@#### ~]# st2 run salt.local_grains.get args=os target=SOMEHOST
`
In salt, I can run something like this to target by grain:
`salt -C "G@kernel:linux and G@os:Redhat and G@virtual:physical" test.ping`

I would like to be able to do the same thing through stackstorm.

2.Any plans on getting rid of the "expr_form" match type? Per the 2019.02 release, Salt has deprecated the match type argument expr_form. Not sure if this may be part of the issue.

**Comment by LindsayHill** [comment]
> Any plans on getting rid of the "expr_form" match type?

This is a user-contributed Open Source pack. There's no specific maintenance other than what users contribute.

No-one else has piped up to say that they're working on an update for that, so submit a PR.

**Comment by nazunalika** [comment]
I have opened a PR for the change: #12 

Thanks chubs

**Comment by nzlosh** [comment]
This issue can be closed as it was resolved by merge #12
